include/eigenpy/details.hpp

```diff
@@ -89,18 +89,19 @@ namespace eigenpy
     static void switchToNumpyArray()
     {
       getInstance().CurrentNumpyType = getInstance().NumpyArrayObject;
-      getInstance().np_type = ARRAY_TYPE;
+      getType() = ARRAY_TYPE;
     }
     
     static void switchToNumpyMatrix()
     {
       getInstance().CurrentNumpyType = getInstance().NumpyMatrixObject;
-      getInstance().np_type = MATRIX_TYPE;
+      getType() = MATRIX_TYPE;
     }
     
-    static NP_TYPE getType()
+    static NP_TYPE & getType()
     {
-      return getInstance().np_type;
+      static NP_TYPE np_type;
+      return np_type;
```

Comment:
Are you sure you solved the issue ?
I think you should move the definition of this function to a non-installed cc file.

Reply:
Both are working of course.